### PR TITLE
Doc: add test specifications into zephyr documentation

### PR DESCRIPTION
**Pull request by enjiamai**
Add test specifications into zephyr doc, this entry will show up in
left side navigation pane, and it contains two categories: integration
and module test specification.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

**Comment by enjiamai** [comment]
Submit this PR due to the original writer leave, the original PR is https://github.com/zephyrproject-rtos/zephyr/pull/29434

File: doc/test_specification/integration_spec/environment.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+When using (at least) one ``-v`` option, sanitycheck's console output
+shows for every test how the test is run (qemu, native_posix, etc.) or
+whether the binary was just built.
+
+To run the script in the local tree, follow the steps below:
+
+::
+
+        $ source zephyr-env.sh
```
Comment (LeiW000) [inline doc/test_specification/integration_spec/environment.rst:32]: As I know, we don't need to do it with the latest code.

**Review by LeiW000 — CHANGES_REQUESTED** [review]
Overall, I think description for integration/module testing has to be re-written to clarify such as what purpose/how we run the test cases, ect.

**Comment by enjiamai** [comment]
> Overall, I think description for integration/module testing has to be re-written to clarify such as what purpose/how we run the test cases, ect.

Got your point, Steven. Will update it as soon.

**Comment by LeiW000** [comment]
@enjiamai , please address the comments.

**Comment by enjiamai** [comment]
I temporarily close this until we start the specification work.

**Pull request closed without merging**